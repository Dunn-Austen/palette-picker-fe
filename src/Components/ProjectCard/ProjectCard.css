.project-card {
  align-items: center;
  background-color: #E4D5D4;
  border: 2px groove purple;
  box-shadow: 0px 2px 10px 1px #52023D;
  display: flex;
  flex-direction: column;
  height: 30vh;
  justify-content: center;
  overflow: scroll;
  width: 40%;
}

.palettes-section {
  align-items: center;
  background-color: #E4D5D4;
  display: flex;
  flex-direction: column;
  height: 30vh;
  justify-content: space-between;
  width: 100%;
}



header {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 5px;
  width: 100%;
}

.title-container {
  display: flex;
  flex-direction: row;
}

.project-title {
  margin: 0;
}

.edit-title {
  height: 30px;
}
