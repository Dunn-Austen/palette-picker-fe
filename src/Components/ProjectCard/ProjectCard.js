import React from 'react';
import './ProjectCard.css';
import PropTypes from 'prop-types';
import PaletteCard from '../PaletteCard/PaletteCard'

const ProjectCard = ({title, palettes, updateProjects, projects}) => {

  const displayPaletteCards = palettes.map((palette, index) => {
    return (
      <PaletteCard
        projects={projects}
        updateProjects={updateProjects}
        title={palette.title}
        id={palette.id}
        project_id={palette.project_id}
        color_1_id={palette.color_1_id}
        color_2_id={palette.color_2_id}
        color_3_id={palette.color_3_id}
        color_4_id={palette.color_4_id}
        color_5_id={palette.color_5_id}
        key={index}
      />
    )
  });

  return (
    <section className='project-card'>
      {displayPaletteCards}
    </section>
  )
}

export default ProjectCard;
