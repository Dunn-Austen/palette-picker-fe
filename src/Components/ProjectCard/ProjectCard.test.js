import React from 'react';
import { shallow } from 'enzyme';
import ProjectCard from './ProjectCard';

describe('ProjectCard', () => {
  it('should match the snapshot', () => {
    const mockTitle = 'Kitchen';
    const mockPalettes = [
      {
        id: 1,
        title: 'counter',
        color_1_id: "#eeeeee",
        color_2_id: "#eeeeee",
        color_3_id: "#eeeeee",
        color_4_id: "#eeeeee",
        color_5_id: "#eeeeee",
        project_id: 1
      }
    ];
    const mockUpdateProjects = jest.fn();
    const mockProjects = [
      {
        id: 1,
        title: 'Kitchen'
      }
    ];
    const wrapper = shallow(<ProjectCard 
      title={mockTitle}
      palettes={mockPalettes}
      updateProjects={mockUpdateProjects}
      projects={mockProjects}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
