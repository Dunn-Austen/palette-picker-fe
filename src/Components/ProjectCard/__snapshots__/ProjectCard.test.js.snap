// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard should match the snapshot 1`] = `
<section
  className="project-card"
>
  <PaletteCard
    color_1_id="#eeeeee"
    color_2_id="#eeeeee"
    color_3_id="#eeeeee"
    color_4_id="#eeeeee"
    color_5_id="#eeeeee"
    id={1}
    key="0"
    project_id={1}
    projects={
      Array [
        Object {
          "id": 1,
          "title": "Kitchen",
        },
      ]
    }
    title="counter"
    updateProjects={[MockFunction]}
  />
</section>
`;
